starky/src/compressor12/compressor12_exec.rs

```diff
@@ -54,10 +55,13 @@ pub fn exec(
         .collect::<Vec<_>>();
 
     for i in 0..adds_len {
-        let w2 = FGL::from_raw_repr(<FGL as PrimeField>::Repr::from(adds[i * 4 + 2])).unwrap();
-        let w3 = FGL::from_raw_repr(<FGL as PrimeField>::Repr::from(adds[i * 4 + 3])).unwrap();
+        // TODO: here we can's assign `let w2 = adds[i].2;`. As adds[i].2 is mont form. But here w2 need mont_reduce form.?
+        let a2: u64 = adds[i].2.into();
+        let a3: u64 = adds[i].3.into();
+        let w2 = FGL::from_raw_repr(<FGL as PrimeField>::Repr::from(a2))?;
```

Comment:
Can you use `let w2 = FGL::from_repr(adds[i].2);` directly?

Reply:
I'll try it.